app/models/notify_user/base_notification.rb

```diff
@@ -211,190 +51,97 @@ def delay_time(options)
     end
 
-    # Used for aggregation when grouping based on group_id for target
-    def self.pending_aggregations_grouped_marked_as_parent(notification)
-      where(type: notification.type)
-      .for_target(notification.target)
-      .where(state: :pending_as_aggregation_parent)
-      .where(group_id: notification.group_id)
+    ## Public Interface
+    def to(user)
+      self.target = user
+      self
     end
 
-    # Used to find all pending notifications with aggregation enabled for target
-    def self.pending_aggregation_by_group_with(notification)
-      for_target(notification.target)
-      .where(state: [:pending, :pending_as_aggregation_parent])
-      .where(group_id: notification.group_id)
+    def with(*args)
+      self.params = args.reduce({}, :update)
+      self
     end
 
-    # Used to find all pending notifications for target
-    def self.pending_aggregation_with(notification)
-      where(type: notification.type)
-      .for_target(notification.target)
-      .where(state: [:pending, :pending_as_aggregation_parent])
+    def grouped_by_id(group_id)
+      self.group_id = group_id
+      self
     end
 
-    def aggregation_pending?
-      # A notification of the same type, that would have an aggregation job associated with it,
-      # already exists.
-
-      # When group aggregation is enabled we provide a different scope
+    # Send any emails / SMS / push notifications:
+    def notify(deliver=true)
       if self.aggregate_grouping
-        return (self.class.pending_aggregations_grouped_marked_as_parent(self).where('id != ?', id).count > 0)
-      else
-        return (self.class.pending_aggregations_marked_as_parent(self).where('id != ?', id).count > 0)
-      end
-    end
-
-    # Aggregates appropriately
-    def deliver
-      if pending? and not user_has_unsubscribed?
-        # if aggregation is false bypass aggregation completely
-        self.channels.each do |channel_name, options|
-          if(options[:aggregate_per] == false)
-            self.mark_as_sent!
-            self.class.delay.deliver_notification_channel(self.id, channel_name)
-          else
-            # only notifies channels if no pending aggregate notifications
-            unless aggregation_pending?
-              self.mark_as_pending_as_aggregation_parent!
-
-              # adds fallback support for integer or array of integers
-              if options[:aggregate_per].kind_of?(Array)
-                self.class.delay_until(delay_time(options)).notify_aggregated_channel(self.id, channel_name)
-              else
-                a_interval = options[:aggregate_per] ? options[:aggregate_per].minutes : self.aggregate_per
-                self.class.delay_for(a_interval).notify_aggregated_channel(self.id, channel_name)
-              end
-            end
-          end
+        group_parents = parents_in_group.where('created_at >= ?', 24.hours.ago).order('created_at DESC')
```

Comment:
Went with what was already there for now, but I was thinking about this as well recently. Could be something nice to add in with little effort. Could even be per channel as opposed to per notification, but either works.